### ISSUE TITLE
Implement hyperparameter optimisation using optuna package

### ISSUE THREAD

**Issue by u7122029** [u7122029/text_gen_calibration#14]
We can, alongside early stopping optimise the following hyperparameters:
- learning rate
- weight decay

For parametric temperature scaling and frequency temperature scaling:
- number of highest tokens
- number of lowest tokens
